src/annotations/parseMetadata.ts

```diff
@@ -0,0 +1,62 @@
+import safeEval from 'safe-eval'
+
+/**
+ * Parse metadata annotations.
+ * 
+ * Examples:
+ *
+ * ```
+ * @metadata
+ * @metadata(create:false, create: "test")
+ * ``
+ *
+ * @param {string} namespace
+ * @param {string?} description
+ * @returns {object}
+ */
+export function parseMetadata(namespace: string, description: string): any {
+  if (description) {
+    const start = `@${namespace}`
+
+    let line = description.split('\n').map(line => line.trim())
+      .find(line => line.startsWith(start))
+    if (!line) {
+      return undefined
+    }
+    line = line.substr(start.length).trim()
+    if (line === '') {
+      return true
+    }
+
+    const openingTag = '('
+    const closingTag = ')'
+
+    if (!line.startsWith(openingTag)) {
+      console.error(`Can't parse "@${namespace} annotation: Expected opening tag "${openingTag}"`)
+    }
+
+    const startPosition = description.indexOf(line)
+    const startContent = description.substring(startPosition + 1, description.length)
+    const endPosition = startContent.indexOf(')')
+
+    if (endPosition === -1) {
+      console.error(`Can't parse "@${namespace} annotation: Expected closing tag "${closingTag}"`)
+
+      return undefined
+    }
+
+    const metadataContent = startContent.substring(0, endPosition).trim()
+    const enclosedContent = `{${metadataContent}}`
+
+    let parsedContent: any;
+    try {
+        parsedContent = safeEval(enclosedContent)
```

Comment:
That is the biggest risk. That will technically execute any code that is enclosed in the directive including functions etc. and return no result. 

I would prefer to have JSON.parse as...
1. Providing content to JSON parse is really well understood by every developer.
2. We can provide reliable error handling and messages

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse

Reply:
I originally tried out JSON.parse, we would just have to wrap every key and string value in "".

```
"""
@db(name: "note_table", columns: ["id", "name"])
"""
```

A direct JSON.parse won't work as this is not valid JSON yet, we would just need to wrap the appropriate parts.

Reply:
What you mean by wrap? Example above is typical instance of the graphql directive.

Reply:
Sorry, I mean we need to enclose the keys `name`, `columns` with "" like `"name"`, `"column"` to make it parsable as valid JSON. Without doing this the above example would be:

`{name: "note_table", columns: ["id", "name"]}` which is JavaScript, not JSON.

Reply:
Yep. `safeEval` FTW